api/provider/aws/deploy_delete.go

```diff
@@ -0,0 +1,55 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
+)
+
+func (d *DeployProvider) Delete(deployID string) error {
+	taskArn, err := d.lookupTaskDefinitionARN(deployID)
+	if err != nil {
+		return err
+	}
+
+	if err := d.deleteDeploy(taskArn); err != nil {
+		return err
+	}
+
+	if err := d.deleteDeployTags(deployID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *DeployProvider) deleteDeploy(taskArn string) error {
+	input := &ecs.DeregisterTaskDefinitionInput{}
+	input.SetTaskDefinition(taskArn)
+
+	if err := input.Validate(); err != nil {
+		return err
+	}
+
+	if _, err := d.AWS.ECS.DeregisterTaskDefinition(input); err != nil {
+		if err, ok := err.(awserr.Error); ok && err.Code() == "does not exist" {
```

Comment:
has this path been tested? I've never seen an aws api code that has spaces in it.

Reply:
The original code is here, it may be cruft, I couldn't repro the behavior:
https://github.com/quintilesims/layer0/blob/develop/api/backend/ecs/deploy_manager.go#L61

Reply:
The ClientException that is thrown when `DeregisterTaskDefinition` can't find the task definition is "The specified task definition does not exist."

Reply:
👍

Reply:
so this will need to be updated to use the `strings.Contains` pattern then